node/pkg/processor/observation.go

```diff
@@ -49,6 +50,11 @@ var (
 func signaturesToVaaFormat(signatures map[common.Address][]byte, gsKeys []common.Address) []*vaa.Signature {
 	// Aggregate all valid signatures into a list of vaa.Signature and construct signed VAA.
 	var sigs []*vaa.Signature
+
+	if len(gsKeys) > math.MaxUint8 {
+		panic(fmt.Sprintf("guardian set too large: %d", len(gsKeys)))
```

Comment:
an explicit `panic` in the guardian innards doesn't make me feel good, but I suppose `gsKeys` being > 255 is a reasonable panic

Reply:
Yeah agreed, maybe we can issue a follow-up PR to resolve this? This function feels like it could be incorporated into the vaa SDK with an error return value anyways.

Reply:
I mean, it shouldn't happen and would be a big deal if it did, so I'm fine with this change as is.